web/docs/guides/websockets.md

```diff
@@ -0,0 +1,129 @@
+---
+title: WebSockets
+---
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+# WebSocket support
+Wasp provides a fully integrated WebSocket experience by utilizing [Socket.IO](https://socket.io/) on the client and server. We handle making sure your URLs are correctly setup, CORS is enabled, and provide a useful `useSocket` abstraction for use in React components.
+
+## Enable in Wasp file
+Here, we specify we are using WebSockets by adding `webSocket` to our `app` and providing the required `fn`. You can optionally specify a list of `entities` for use in your context (like in operations).
+
+```wasp title=todoApp.wasp
+app todoApp {
+  // ...
+
+  webSocket: {
+    fn: import { webSocketFn } from "@server/webSocket.js"
+    // Optional: entities: [...]
+  },,
+}
+```
+
+## Optional: Add shared types for client and server use
+If TypeScript is your thing, you can add shared interface definitions to use on both the client and server. This allows you to type all the messages you send and receive!
+
+```ts title=src/shared/webSocket.ts
+import { WaspSocketData } from '@wasp/universal/types'
+
+export interface ServerToClientEvents {
+  chatMessage: (msg: { id: string, username: string, text: string }) => void;
+}
+
+export interface ClientToServerEvents {
+  chatMessage: (msg: string) => void;
+}
+
+export interface InterServerEvents { }
+
+// Data that is attached to the socket.
+// NOTE: Wasp automatically injects the JWT into the connection,
```

Comment:
Saying `JWT` here feels bit low level, I would maybe say "auth info" or something like that so people don't need to think too hard.